test: setup Jest and add unit test for PlaceCard

kimmie-ngo opened the pull request.

kimmie-ngo committed:
test: setup Jest and add unit test for PlaceCard

notyalcs commented on src/components/NearbySportsFinder.tsx:
```diff
@@ -47,11 +47,11 @@ const NearbySportsFinder: React.FC = () => {
 
             //localhost development
-            // const response = await axios.get(`http://localhost:5102/locations`);
+             const response = await axios.get(`http://localhost:5102/locations`);
```
You have to switch this back

kimmie-ngo replied:
yes, always forget to switch back

notyalcs commented on src/pages/LocationsPage.tsx:
```diff
@@ -15,10 +15,10 @@ const LocationsPage = () => {
 
       //localhost development
-      //  const response = await axios.get(`http://localhost:5102/locations`);
+      const response = await axios.get(`http://localhost:5102/locations`);
```
you have to switch this

notyalcs reviewed (changes requested):
The axios calls are switched to local. Please switch them back.

kimmie-ngo committed:
switched to online env #50 #53

kimmie-ngo committed:
switched to online env #18 #21

notyalcs reviewed (approved):
LG2M

kimmie-ngo committed:
new tests for functional requirements

kimmie-ngo committed:
Adding tests

kimmie-ngo committed:
update to use ApiBaseUrl
replace import.meta.env with Vite-defined global constant for test compatibility

kimmie-ngo committed:
fix: add Node types to tsconfig.app.json to resolve process.env error during build

kimmie-ngo reviewed (commented):
Please merge this branch after you check and approve, Sean/Han

notyalcs commented on src/components/NearbySportsFinder.tsx:
```diff
@@ -46,12 +47,7 @@ const NearbySportsFinder: React.FC = () => {
-            //online deployment
-            const response = await axios.get(`${import.meta.env.VITE_BACKEND_URL}/locations`);
-            console.log("API response:", response.data);
+            const response = await axios.get(`${getApiBaseUrl()}/locations`);
```
I don't think this will work with the new endpoint of `api/Composite/locations`. Can you merge the main branch into this and resolve the conflicts and make sure it works with microservices

notyalcs commented on src/pages/LocationsPage.tsx:
```diff
@@ -14,11 +15,7 @@ const LocationsPage = () => {
-
-      //localhost development
-      //  const response = await axios.get(`http://localhost:5102/locations`);
+        const response = await axios.get(`${getApiBaseUrl()}/locations`);
```
Same comment as for NearbySportsFinder.tsx

notyalcs commented on src/utils/getApiBaseUrl.ts:
```diff
@@ -0,0 +1,9 @@
+
+  // ✅ At build time, Vite will replace this string literal
+  return __APP_BACKEND_URL__;
+};
```
Does this work with there being multiple backend urls now?

notyalcs commented on vite.config.ts:
```diff
@@ -38,4 +38,7 @@ export default defineConfig({
     }),
   ],
+  define: {
+    __APP_BACKEND_URL__: JSON.stringify(process.env.VITE_BACKEND_URL || 'http://localhost:5102'),
```
I think this will cause a problem with the microservices

notyalcs reviewed (changes requested):
This branch doesn't have microservice support merged into it. Before fixing the endpoint logic, you should merge Main back into this branch I think
